File: src/Core/Extensions/StreamExtensions.cs
using System.IO;

namespace NPackage.Core.Extensions
{
    public static class StreamExtensions
    {
        public static void CopyTo(this Stream inputStream, string outputFilename)
        {
            using (Stream outputStream = File.Create(outputFilename))
            {
                int count;
                byte[] chunk = new byte[4096];
                while ((count = inputStream.Read(chunk, 0, chunk.Length)) > 0)
                    outputStream.Write(chunk, 0, count);
            }
        }
    }
}

File: src/Core/StreamExtensions.fs
module NPackage.Core.StreamExtensions

open System.IO

let rec copyWith (buffer : array<byte>) (inputStream : Stream) (outputStream : Stream) =
    match inputStream.Read(buffer, 0, buffer.Length) with
    | count when count > 0 ->
        outputStream.Write(buffer, 0, count)
        copyWith buffer inputStream outputStream

    | _ -> ()

let copy inputStream outputStream =
    copyWith (Array.zeroCreate 4096) inputStream outputStream
